core/src/main/java/com/linecorp/armeria/internal/common/metric/RequestMetricSupport.java

```diff
@@ -161,6 +162,27 @@ private static void updateMetrics(
         }
     }
 
+    private static void updateRetryingClientMetrics(
+            RequestContext ctx, RequestLog log,
+            ClientRequestMetrics metrics, int childrenSize,
+            SuccessFunction successFunction) {
+        int successAttempts = 0;
+        int failureAttempts = 0;
+
+        metrics.actualRequests().increment(childrenSize);
+
+        for (RequestLogAccess child : log.children()) {
+            if (successFunction.isSuccess(ctx, child.ensureComplete())) {
+                successAttempts++;
+            } else {
+                failureAttempts++;
+            }
+        }
+
+        metrics.successAttempts().record(successAttempts);
+        metrics.failureAttempts().record(failureAttempts);
```

Comment:
I was rather imaging that `successFunction` would be applied to the overall response once, and the number of attempts is logged.
The reasoning is that `RetryRule` doesn't necessarily coincide with whether a response is successful or not.

We probably don't want `successFunction` to be evaluated multiple times, so we may need some refactoring
```
        boolean success = successFunction.isSuccess(ctx, log);
        if (success) {
            metrics.successAttempts().record(log.children().size());
        } else {
            metrics.failureAttempts().record(log.children().size());
        }
```